### PR TITLE
Remove kubernetes 1.20 variants

### LINKED ISSUE #2400: Deprecate K8s 1.20 variants by November 2022
Kubernetes 1.20 went out of upstream support in December, 2021. AWS EKS is maintaining support for 1.20 until [November 1st, 2022](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#kubernetes-release-calendar). Bottlerocket will deprecate support of our K8s 1.20 variants at that time.

While you’ll still be able to launch `*-k8s-1.20` nodes beyond November 1st, 2022, we’ll no longer provide updated `*-k8s-1.20` images, and the Bottlerocket repositories will no longer receive new updates for nodes running K8s 1.20 variants.

We recommend you replace existing Bottlerocket K8s 1.20 nodes with [a newer variant](https://github.com/bottlerocket-os/bottlerocket/tree/develop/variants#variants) - ideally the latest your Kubernetes cluster supports.

For instructions on how to migrate your `aws-k8s-1.20` nodes to newer supported versions, please consult the [EKS documentation for updating clusters](https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html) or reach out to us through your AWS support channels for additional questions.

### PR DESCRIPTION
**Pull request by jpmcb**
**Issue number:**

Closes #2400
Reference PR: #2316

**Description of changes:**

```
- Remove aws-k8s-1.20 variant
- Remove vmware-k8s-1.20 variant

Signed-off-by: John McBride <jpmmcb@amazon.com>
```

**Testing done:**

`cargo make check && cargo make build` 👍🏼 

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by jpmcb** [comment]
Marking this draft until we're abit closer to the November timeline (when EKS support for k8s 1.20 will drop per the issue noted above)

**Comment by jpmcb** [comment]
Ready for review now and will target the `1.11.0` release

File: PUBLISHING-VMWARE.md [hunk 1]
```diff
@@ -46,20 +46,20 @@ datacenters = ["foo", "bar"]
 Then you can easily upload your OVA, specifying the variant you wish to upload (currently only VMware variants).
 
 ```shell
-cargo make -e BUILDSYS_VARIANT=vmware-k8s-1.20 upload-ova
+cargo make -e BUILDSYS_VARIANT=vmware-k8s-1.21 upload-ova
```
Comment (etungsten) [inline PUBLISHING-VMWARE.md]: Let's move this to the latest.
```suggestion
cargo make -e BUILDSYS_VARIANT=vmware-k8s-1.24 upload-ova
```

**Review by etungsten — CHANGES_REQUESTED** [review]
We should delete `models/src/aws-k8s-1.20` like in https://github.com/bottlerocket-os/bottlerocket/pull/2316/files#.

**Comment by jpmcb** [comment]
Force pushed to also remove `sources/models/src/aws-k8s-1.20` and `sources/models/src/vmware-k8s-1.20`. Updated `PUBLISHING-VMWARE.md` to indicate using 1.24 kubernetes

**Review by webern — APPROVED** [review]
I'm not sure the best way to make sure nothing is missed when reviewing this type of PR. I did compare it to https://github.com/bottlerocket-os/bottlerocket/pull/2316 though and didn't surface any issues.

**Review by etungsten — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -77,7 +76,6 @@ The following variants support ECS:
 
 We also have variants that are designed to be Kubernetes worker nodes in VMware:
 
-* `vmware-k8s-1.20`
```
Comment (bcressey) [inline README.md:80]: Traditionally we add the removed variants to the "The following variants are no longer supported" section below.

**Review by bcressey — APPROVED** [review]

**Comment by jpmcb** [comment]
Force pushed to address Ben's comment: Added `aws-k8s-1.20` and `vmware-k8s-1.20` to the "The following variants are no longer supported" section. Will merge with passing tests.

**Review by bcressey — APPROVED** [review]

**Review by stmcginnis — APPROVED** [review]
image

🥇

**Pull request merged**